ui/admin/app/components/form/credential-library/vault-ssh-cert/index.hbs

```diff
@@ -1,2 +1,287 @@
+  </Hds::Form::Textarea::Field>
+
+  {{#if (and @model.isNew (feature-flag 'credential-library-vault-ssh-cert'))}}
+    <Hds::Form::RadioCard::Group @name='type' @alignment='center' as |G|>
+      <G.Legend>{{t 'form.type.label'}}</G.Legend>
+      {{#each this.types as |type|}}
+        <G.RadioCard
+          @value={{type}}
+          @maxWidth='20rem'
+          @checked={{eq type @model.type}}
+          {{on 'input' (fn @changeType type)}}
+          as |R|
+        >
+          <R.Label>{{t
+              (concat 'resources.credential-library.types.' type)
+            }}</R.Label>
+          <R.Description>{{t
+              (concat 'resources.credential-library.help.' type)
+            }}</R.Description>
+        </G.RadioCard>
+      {{/each}}
+    </Hds::Form::RadioCard::Group>
+  {{else}}
+    <InfoField
+      @value={{t (concat 'resources.credential-library.types.' @model.type)}}
+      as |F|
+    >
+      <F.Label>{{t 'form.type.label'}}</F.Label>
+      <F.HelperText>
+        {{t (concat 'resources.credential-library.help.' @model.type)}}
+      </F.HelperText>
+    </InfoField>
+  {{/if}}
+
+  <Hds::Form::TextInput::Field
+    @isRequired={{true}}
+    @value={{@model.path}}
+    @isInvalid={{@model.errors.path}}
+    @type='text'
+    name='vault_path'
+    disabled={{form.disabled}}
+    {{on 'input' (set-from-event @model 'path')}}
+    as |F|
+  >
+    <F.Label>{{t
+        'resources.credential-library.form.vault_path.label'
+      }}</F.Label>
+    <F.HelperText>{{t
+        'resources.credential-library.form.vault_path.help'
+      }}</F.HelperText>
+    {{#if @model.errors.path}}
+      <F.Error as |E|>
+        {{#each @model.errors.path as |error|}}
+          <E.Message>{{error.message}}</E.Message>
+        {{/each}}
+      </F.Error>
+    {{/if}}
+  </Hds::Form::TextInput::Field>
+
+  <Hds::Form::TextInput::Field
+    @isRequired={{true}}
+    @value={{@model.username}}
+    @isInvalid={{@model.errors.username}}
+    @type='text'
+    name='username'
+    disabled={{form.disabled}}
+    {{on 'input' (set-from-event @model 'username')}}
+    as |F|
+  >
+    <F.Label>{{t 'resources.credential-library.form.username.label'}}</F.Label>
+    <F.HelperText>{{t
+        'resources.credential-library.form.username.help'
+      }}</F.HelperText>
+    {{#if @model.errors.username}}
+      <F.Error as |E|>
+        {{#each @model.errors.username as |error|}}
+          <E.Message>{{error.message}}</E.Message>
+        {{/each}}
+      </F.Error>
+    {{/if}}
+  </Hds::Form::TextInput::Field>
+
+  <Hds::Form::Select::Field
+    @isRequired={{true}}
+    @width='100%'
+    name='key_type'
+    disabled={{form.disabled}}
+    {{on 'input' (set-from-event @model 'key_type')}}
+    as |F|
+  >
+    <F.Label>{{t 'resources.credential-library.form.key_type.label'}}</F.Label>
+    <F.HelperText>{{t
+        'resources.credential-library.form.key_type.help'
+      }}</F.HelperText>
+    <F.Options>
+      <option value=''></option>
```

Comment:
What's the purpose of this item?  Should this at least have visible text?

Reply:
It's to show an empty option so a user actually has to select it. 

There's a somewhat problematic UX where without the option, the model itself doesn't have a value but the select just displays the first option and it doesn't get saved but it _looks_ like it did even if you viewed the form again. 

I would have to set a default option on the model to get around this since we don't set anything until the user triggers the `onInput` which I'm open to, but I thought this might be better since a user has to make a conscious decision. 

I'm not sure if it's a common pattern/UX but the code examples in [HDS](https://helios.hashicorp.design/components/form/select?tab=code) show the usage of them.

Reply:
Should it say "Choose a type" or something?

Reply:
Yup that makes sense, I'll update it